src/app/dataset-block/metadata-block/components/event-details/event-details.component.ts

```diff
@@ -1,118 +1,116 @@
-    }
-
-    public get isSetTransformEvent(): boolean {
-        return this.currentBlock.event.__typename === SupportedEvents.SetTransform;
-    }
-
-    public get setTransformEvent(): SetTransform {
-        return this.currentBlock.event as SetTransform;
-    }
-
-    public get setPollingSourceEvent(): SetPollingSource {
-        return this.currentBlock.event as SetPollingSource;
-    }
-
-    public get addDataEvent(): AddDataEventFragment {
-        return this.currentBlock.event as AddDataEventFragment;
-    }
+    @ViewChild("dynamicContainer", { read: ViewContainerRef })
+    public dynamicContainer: MaybeNull<ViewContainerRef>;
 
-    public get isSeedEvent(): boolean {
-        return this.currentBlock.event.__typename === SupportedEvents.Seed;
-    }
-
-    public get seedEvent(): Seed {
-        return this.currentBlock.event as Seed;
-    }
-
-    public get isExecuteQueryEvent(): boolean {
-        return this.currentBlock.event.__typename === SupportedEvents.ExecuteQuery;
-    }
-
-    public get executeQueryEvent(): ExecuteQuery {
-        return this.currentBlock.event as ExecuteQuery;
-    }
-
-    public get isSetLicenseEvent(): boolean {
-        return this.currentBlock.event.__typename === SupportedEvents.SetLicense;
-    }
-
-    public get setLicenseEvent(): SetLicense {
-        return this.currentBlock.event as SetLicense;
-    }
-
-    public get isSetAttachmentsEvent(): boolean {
-        return this.currentBlock.event.__typename === SupportedEvents.SetAttachments;
-    }
-
-    public get setAttachmentsEvent(): SetAttachments {
-        return this.currentBlock.event as SetAttachments;
-    }
-
-    public get isSetInfoEvent(): boolean {
-        return this.currentBlock.event.__typename === SupportedEvents.SetInfo;
-    }
-
-    public get setInfoEvent(): SetInfo {
-        return this.currentBlock.event as SetInfo;
-    }
-
-    public get isSetVocabEvent(): boolean {
-        return this.currentBlock.event.__typename === SupportedEvents.SetVocab;
-    }
-
-    public get setVocabEvent(): SetVocab {
-        return this.currentBlock.event as SetVocab;
-    }
+    constructor(private blockService: BlockService) {}
 
-    public get isSetWatermarkEvent(): boolean {
-        return this.currentBlock.event.__typename === SupportedEvents.SetWatermark;
+    ngAfterViewChecked(): void {
+        if (this.dynamicContainer) {
+            this.dynamicContainer.clear();
+            const componentRef = this.dynamicContainer.createComponent<BaseComponent>(
+                this.componentEventTypeFactory(this.currentBlock.event.__typename as SupportedEvents),
+            );
+            componentRef.setInput("event", this.currentBlock.event);
+            componentRef.changeDetectorRef.detectChanges();
+        }
+    }
+
+    private componentEventTypeFactory(type: SupportedEvents): Type<BaseComponent> {
+        let component: Type<BaseComponent>;
+        switch (type) {
```

Comment:
A syntax based on dictionary might be more compact than switch statement.

Reply:
Changed to dictionary